Makefile

```diff
@@ -0,0 +1,93 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# -------------------------------------------------------------
+# This makefile defines the following targets
+#
+#   - peer (default) - builds the fabric ./peer/peer binary
+#   - membersrvc - builds the ./membersrvc/membersrvc binary
+#   - unit-test - runs the go-test based unit tests
+#   - behave - runs the behave test
+#   - behave-deps - ensures pre-requisites are availble for running behave manually
+#   - images - ensures all docker images are available
+#   - peer-image - ensures the peer-image is available (for behave, etc)
+#   - ca-image - ensures the ca-image is available (for behave, etc)
+#   - protos - generate all protobuf artifacts based on .proto files
+#   - clean - cleans the build area
+#   - dist-clean - superset of 'clean' that also removes persistent state
+
+
+PKGNAME = github.com/hyperledger/fabric
+CGO_LDFLAGS = -lrocksdb -lstdc++ -lm -lz -lbz2 -lsnappy
+
+EXECUTABLES = go docker
+K := $(foreach exec,$(EXECUTABLES),\
+	$(if $(shell which $(exec)),some string,$(error "No $(exec) in PATH: Check dependencies")))
+
+all: peer
```

Comment:
I would think that all should build peer, unit-test and behave - no? Should we also run the go tools as part of the make, as well?